### PR TITLE
better lda edit page; lda info repo

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem @eishay 
- created a new repo.
- factor out a ldaInfoCommander out of old LDACommander. 
- new admin page

todo soon: 
- a scheduled job to query for num of docs in a topic, and persist it in the new repo
- admin to config the 'isNameable' field. currently everything defaults to 'true'

### CODE REVIEW

File: kifi-backend/modules/common/conf/evolutions/shoebox/222.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version tinyint unsigned NOT NULL,
+  dimension smallint unsigned NOT NULL,
+  topic_id smallint unsigned NOT NULL,
+  topic_name varchar(64) NOT NULL,
+  is_active bool NOT NULL,
+  is_nameable bool NOT NULL,
+  num_docs int unsigned NOT NULL,
+
+  PRIMARY Key (id),
+  INDEX lda_info_i_version_topic_id (version, topic_id)
```
Comment (eishay) [inline kifi-backend/modules/common/conf/evolutions/shoebox/222.sql]: unique?

File: kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/LDAInfoRepo.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  def getDimension(version: ModelVersion[DenseLDA])(implicit session: RSession): Option[Int] = {
+    (for { r <- rows if r.version === version } yield r).firstOption.map { _.dimension }
+  }
+
+  def getAllByVersion(version: ModelVersion[DenseLDA])(implicit session: RSession): Seq[LDAInfo] = {
+    (for { r <- rows if r.version === version } yield r).list
+  }
+
+  def getUnamed(version: ModelVersion[DenseLDA], limit: Int)(implicit session: RSession): Seq[LDAInfo] = {
+    (for { r <- rows if r.version === version && r.topicName === LDAInfo.DEFUALT_NAME && r.isNameable === true } yield r).sortBy(_.numOfDocs.desc).list.take(limit)
```
Comment (eishay) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/dbmodel/LDAInfoRepo.scala]: take before list

File: kifi-backend/modules/cortex/test/com/keepit/cortex/dbmodel/LDAInfoRepoTest.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          repo.getDimension(version).get === 4
+        }
+      }
+    }
+
+    "query unamed topics" in {
+      withDb() { implicit injector =>
+        val repo = inject[LDAInfoRepo]
+        db.readWrite { implicit s =>
+          List(info1, info2, info3, info4).foreach { repo.save(_) }
```
Comment (eishay) [inline kifi-backend/modules/cortex/test/com/keepit/cortex/dbmodel/LDAInfoRepoTest.scala:31]: `info1 :: info2 :: info3 :: info4 :: Nil`

**Comment by yingjieMiao** [comment]
@stkem   discussed with @eishay in person. Feel free to comment later on.

**Pull request merged**